todo_app/data/TrelloSession.py

```diff
@@ -0,0 +1,65 @@
+"""
+    TrelloSession.py
+
+
+"""
+import urllib.parse as UP
+import requests
+
+# Handle Corporate IT wonk. TODO once I'm back in the office
+PROXIES={}
+REQUESTS_VERIFY=True
+
+
+class TrelloSession(object):
+    """ Class to handle the dialect of REST used by trello.
+
+        In particular, it encapsulates how to pass authentication information to
+        the server, and returns response body JSON.
+    """
+    def __init__(self, root_url, api_key, token):
+        self._root_url = root_url
+        self._api_key = api_key
+        self._token = token
+
+    def request_url(self, path, query_dict=None):
+        """ Encapsulate the wonk for the Marvel API authentication.
+            query_dict contains the arguments for the query, which are merged
+            into the needed auth query items. 
+        """
+        if query_dict is None:
+            query_dict={}

```

Comment:
You set a default query_dict here then immediately override it - would this be nicer as:
```suggestion
    def request_url(self, path, query_dict={}):
        """ Encapsulate the wonk for the Marvel API authentication.
            query_dict contains the arguments for the query, which are merged
            into the needed auth query items. 
        """
```

Reply:
That said, see my other comment re handling these just directly as params

Reply:
I considered commenting this idiom as I thought a reviewer might baulk at it.  It's a style I converged on because of the following issue: 

```
def func_a(mandatory, optional={}):
   # do something dict-y with optional
...
def func_b(mandatory1, mandatory2, optional1=None, optional2=None):
    ...
    func_a( mandatory1, optional2)
```
so when func_b is allowed to default, func_a gets a wrong "default" value.   Now I have to go and sit down and think whether **kwargs solves this...

Oooh, deep learnings!